kedro/contrib/io/matplotlib/matplotlib_s3_writer.py

```diff
@@ -0,0 +1,162 @@
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#
+# The QuantumBlack Visual Analytics Limited ("QuantumBlack") name and logo
+# (either separately or in combination, "QuantumBlack Trademarks") are
+# trademarks of QuantumBlack. The License does not grant you any right or
+# license to the QuantumBlack Trademarks. You may not use the QuantumBlack
+# Trademarks or any confusingly similar mark as a trademark for your product,
+#     or use the QuantumBlack Trademarks in any other manner that might cause
+# confusion in the marketplace, including but not limited to in advertising,
+# on websites, or on software.
+#
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""
+``MatplotlibWriterS3`` saves matplotlib objects as image files to s3.
+"""
+
+import copy
+import io
+from typing import Any, Dict, List, Optional, Union
+
+from matplotlib.pyplot import figure
+from s3fs import S3FileSystem
+
+from kedro.io import AbstractDataSet, DataSetError
+
+
+class MatplotlibWriterS3(AbstractDataSet):
+    # pylint: disable=too-many-instance-attributes
+    """
+    ``MatplotlibWriter`` saves matplotlib objects as image files.
+
+    Example:
+    ::
+
+        import matplotlib.pyplot as plt
+        from kedro.contrib.io.matplotlib_s3_writer import MatplotlibWriterS3
+
+        # Saving single plot
+        plt.plot([1, 2, 3], [4, 5, 6])
+        single_plot_writer = MatplotlibWriterS3(
+            bucket="my-super-great-bucket", filepath="matplot_lib_single_plot.png"
+        )
+        single_plot_writer.save(plt)
+
+        # Saving dictionary of plots (with SSE)
+        plots_dict = {}
+        for colour in ["blue", "green", "red"]:
+            plots_dict[colour] = plt.figure()
+            plt.plot([1, 2, 3], [4, 5, 6], color=colour)
+            plt.close()
+        dict_plot_writer = MatplotlibWriterS3(
+            bucket="my-super-great-bucket", filepath="matplotlib_dict",
+        )
+        dict_plot_writer.save(plots_dict)
+
+        # Saving list of plots
+        plots_list = []
+        for index in range(5):
+            plots_list.append(plt.figure())
+            plt.plot([1,2,3],[4,5,6], color=colour)
+        list_plot_writer = MatplotlibWriterS3(
+            bucket="my-super-great-bucket", filepath="matplotlib_list"
+        )
+        list_plot_writer.save(plots_list)
+    """
+
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        bucket_name: str,
+        filepath: str,
+        s3fs_args: Optional[Dict] = None,
+        credentials: Optional[Dict[str, Any]] = None,
+        savefig_args: Dict[str, Any] = None,
+    ) -> None:
+        """Creates a new instance of ``MatplotlibWriter``.
+
+        Args:
+            bucket_name: Name of the bucket without "s3://" prefix
+            filepath: Path to a matplot object file.
+            s3fs_args: Dictionary of arguments
```

Comment:
Not very clear where these go, a link to the `S3FileSystem` docs would be useful, or a mention that this is where you'd specify the encryption type.